std_execution.bs

```diff
@@ -1247,6 +1248,133 @@ talk about the semantics of some of the algorithms proposed in this paper.
 
 This paper will be updated in the future with any changes that arise from the investigations into P2175R0.
 
+## Sender factories and adaptors are lazy ## {#design-lazy-algorithms}
+
+In an earlier revision of this paper, some of the proposed algorithms supported
+executing their logic eagerly; <i>i.e.</i>, before the returned sender has been
+connected to a receiver and started. These algorithms were removed because eager
+execution has a number of negative semantic and performance implications.
+
+### Eager execution leads to detached work or worse ### {#design-lazy-algorithms-detached}
+
+One of the questions that arises with APIs that can potentially return
+eagerly-executing senders is "What happens when those senders are destructed
+without a call to `execution::connect`?" or similarly, "What happens if a call
+to `execution::connect` is made, but the returned operation state is destroyed
+before `execution::start` is called on that operation state"?
+
+In these cases, the operation represented by the sender is potentially executing
+concurrently in another thread at the time that the destructor of the sender
+and/or operation-state is running. In the case that the operation has not
+completed executing by the time that the destructor is run we need to decide
+what the semantics of the destructor is.
+
+There are three main strategies that can be adopted here, none of which is
+particularly satisfactory:
+
+1.  Make this undefined-behaviour - the caller must ensure that any
+    eagerly-executing sender is always joined by connecting and starting that
+    sender. This approach is generally pretty hostile to programmers,
+    particularly in the presence of exceptions, and is thus not recommended.
```

Comment:
I would probably expand on this a little and say that it is hostile to the ability to compose these operations.
If I pass ownership of an eager sender to another algorithm's sender then I need to make sure both that I connect/join that sender and that the algorithm I passed the sender to also guarantees to connect/join the eager sender if I connect/join that algorithm's sender.
This can be problematic if composing an eager sender with senders that might fail on `connect()` (e.g. because they allocate resources in `connect()` which might fail) or if composing with other eager senders.
e.g. `when_all(eager_op_1(), eager_op_2())`

As eager operations typically need to allocate resources on the heap in order to start the operations, calling an eager algorithm can potentially throw an exception, so in this case calling `eager_op_1()` might start an eager op and then `eager_op_2()` throws an exception. This was the basic problem that I described in my CppCon talk.